packages/forms/src/UIForm/fields/InputDateTimePicker/InputDateTimePicker.component.js

```diff
@@ -0,0 +1,190 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import memoize from 'lodash/memoize';
+import InputDateTimePickerComponent from '@talend/react-components/lib/DateTimePickers';
+import FieldTemplate from '../FieldTemplate';
+import { isoDateTimeRegExp } from '../../customFormats';
+import { UnhandleTypeError, UnexpectedTypeError } from './WrongTypeError';
+
+export const GENERIC_FORMAT_ERROR = 'GENERIC FORMAT ERROR';
+
+const INVALID_DATE = new Date('');
+const HANDLE_CONVERTION_TYPE = ['string', 'number'];
+
+function isDateValid(date) {
+	if (date === undefined) {
+		return true;
+	}
+
+	return date instanceof Date && !isNaN(date.getTime());
+}
+
+function convertDateToTimestamp(date) {
+	return date.getTime();
+}
+
+function convertTimestampToDate(timestamp) {
+	return new Date(timestamp);
+}
+
+function convertDateToString(date) {
+	return date.toISOString();
+}
+
+function convertStringToDate(str) {
+	if (!isoDateTimeRegExp.test(str)) {
+		return INVALID_DATE;
+	}
+
+	return new Date(str);
+}
+
+function convertToDate(type, value) {
+	if (value === undefined) {
+		return undefined;
+	}
+
+	const typeOfValue = typeof value;
+
+	if (typeOfValue !== type) {
+		// eslint-disable-next-line no-console
+		console.error(new UnexpectedTypeError(type, typeOfValue));
+		return INVALID_DATE;
+	}
+
+	switch (type) {
+		case 'number':
+			return convertTimestampToDate(value);
+		case 'string':
+			return convertStringToDate(value);
+		default:
+			// eslint-disable-next-line no-console
+			console.error(new UnhandleTypeError(HANDLE_CONVERTION_TYPE, type));
+			return INVALID_DATE;
+	}
+}
+
+function convertFromDate(type, date) {
+	if (date === undefined) {
+		return undefined;
+	}
+
+	switch (type) {
+		case 'number':
+			return convertDateToTimestamp(date);
+		case 'string':
+			return convertDateToString(date);
+		default: {
+			const unhandleTypeError = new UnhandleTypeError(HANDLE_CONVERTION_TYPE, type);
+			// eslint-disable-next-line no-console
+			console.error(unhandleTypeError);
+			return unhandleTypeError;
+		}
+	}
+}
+
+class InputDateTimePicker extends React.Component {
+	constructor(props) {
+		super(props);
+
+		this.state = {
+			value: props.value,
+		};
+
+		this.onChange = this.onChange.bind(this);
+		this.onBlur = this.onBlur.bind(this);
+		this.convertToDate = memoize(convertToDate, (type, value) => `${type}||${value}`);
+	}
+
+	componentWillReceiveProps(nextProps) {
```

Comment:
Because deconstructing inside the arguments function definition hide the semantic. With the name argument we know for sure, it's not some props (last one, new one, which one ?) but **next** props.